secret/secret.go

```diff
@@ -64,48 +68,147 @@ func UpdateSecrets(s secretInterface, secrets *v1.Secret, create bool) {
 		if err != nil {
 			logFatal(err)
 		}
-		log.Println("Created `secret`")
+		log.Printf("Created `%s`\n", secrets.Name)
 	} else if util.IsDirty(secrets) {
 		util.MarkAsClean(secrets)
 		_, err := s.Update(secrets)
 		if err != nil {
 			logFatal(err)
 		}
-		log.Println("Updated `secret`")
+		log.Printf("Updated `%s`\n", secrets.Name)
+	}
+}
+
+func FindSecret(s secretInterface) (*v1.Secret, error) {
```

Comment:
I think the logic in this function is too complicated. There are only 3 valid cases:

1. The previous secrets are called "$SECRET_NAME-$(revision-1)", or

2. the previous secrets are called `$SECRET_NAME`, or

3. there are no previous secrets; this is an install and not an upgrade.

If the previous revision secrets don't exist, then we should not look for even older versions; but throw an error. This situation can only happen when somebody manually removed the previous secrets.

So this function should probably renamed `FindPreviousSecrets(revision int)` or something like that.

Reply:
Yes, accepting only R-1 and without R simplifies this much.

Reply:
Will do.